### PR TITLE
www: rewriting "data fetching" code example

### PR DESCRIPTION
**Pull request by Asqit**
Creating pull request on issue #1553

### CODE REVIEW

File: docs/latest/getting-started/fetching-data.md [hunk 1]
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
...
 user data from an API or database.
 
-These operations are all asynchronous. Rendering however, is always synchronous.
-Instead of fetching data directly during rendering, it should be loaded in a
-route's `handler` function and then passed to the page component via first
-argument to `ctx.render()`.
+These operations are all asynchronous. However, rendering is always synchronous.
+So, to fetch data dynamically, we can use our brand new `async-route`
+components, which abstract the old way of fetching data by writing the
+`Handlers`.
```
Comment (iuioiua) [inline docs/latest/getting-started/fetching-data.md]: This entire section may not be needed now.

File: docs/latest/getting-started/fetching-data.md [hunk 2]
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
...
+parameter is mandatory, and it indicates that we will fetch some data. Then we
+need the `ctx` prop to get our route parameters (in this case, it's the
+username).
 
 Here is an example of a route that fetches user data from the GitHub API and
 renders it in a page component.
 
 ```tsx
-// routes/github/[username].tsx
+// File: 'routes/github/[username].tsx'
```
Comment (iuioiua) [inline docs/latest/getting-started/fetching-data.md]:
```suggestion
// routes/github/[username].tsx
```
I think it's obvious enough that the commented path refers to the file in question. It's ok to leave the `File:` prefix out.

File: docs/latest/getting-started/fetching-data.md [hunk 3]
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
...
+components, which abstract the old way of fetching data by writing the
+`Handlers`.
 
-The data that is passed to `ctx.render()` can then be accessed via the
-`props.data` field on the page component.
+To fetch our data, we must first make our route component asynchronous and then
+create a way to fetch the data, for example with the `fetch` API. The `req`
+parameter is mandatory, and it indicates that we will fetch some data. Then we
+need the `ctx` prop to get our route parameters (in this case, it's the
+username).
```
Comment (iuioiua) [inline docs/latest/getting-started/fetching-data.md]: This page is not a guide on how to refactor the old handlers to new async route components. Can you rewrite this so it doesn't sound like one?

File: docs/latest/getting-started/fetching-data.md [hunk 4]
```diff
@@ -9,56 +9,53 @@ during rendering. In real projects, this is often different. In many cases you
...
-    const { username } = ctx.params;
-    const resp = await fetch(`https://api.github.com/users/${username}`);
-    if (resp.status === 404) {
-      return ctx.render(null);
-    }
-    const user: User = await resp.json();
-    return ctx.render(user);
-  },
-};
+export default async function Page(req: Request, ctx: RouteContext) {
```
Comment (iuioiua) [inline docs/latest/getting-started/fetching-data.md]:
```suggestion
export default async function Page(_req: Request, ctx: RouteContext) {
```
As the `Request` object isn't being used, we prepend it with an `_`.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Great work! Just needs a bit of tweaking.

File: docs/latest/getting-started/fetching-data.md (hunk 3 above)
Comment (Asqit) [reply docs/latest/getting-started/fetching-data.md]: Thank you! Does this sound better? 

"
In order to fetch data, the route component must be asynchronous. The `req`
parameter is mandatory, and it indicates that certain data will be fetched. The
second `req` parameter is used to get the route parameters, such as the username
is this case.
"

**Review by iuioiua — APPROVED** [review]
I've made a few tweaks. Now, LGTM! Thanks for helping out, @Asqit.

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**